imsim/skycat.py

```diff
@@ -0,0 +1,355 @@
+        """
+        Return the sky coordinates of the skyCatalog object
+        corresponding to the specified index.
+
+        Parameters
+        ----------
+        index : int
+            Index of the (object_index, subcomponent) combination.
+
+        Returns
+        -------
+        galsim.CelestialCoord
+        """
+        skycat_obj = self.objects[index]
+        ra, dec = skycat_obj.ra, skycat_obj.dec
+        return galsim.CelestialCoord(ra*galsim.degrees, dec*galsim.degrees)
+
+    def getLens(self, skycat_obj):
+        """
+        Return the weak lensing parameters for the skyCatalog object
+        corresponding to the specified index.
+
+        Parameters
+        ----------
+        skycat_obj : skycatalogs.BaseObject
+            The skycatalogs object, e.g., star or galaxy, which can
+            have several subcomponents, e.g., disk, bulge, knots.
+
+        Returns
+        -------
+        (g1, g2, mu)
+        """
+        gamma1 = skycat_obj.get_native_attribute('shear_1')
+        gamma2 = skycat_obj.get_native_attribute('shear_2')
+        kappa =  skycat_obj.get_native_attribute('convergence')
+        # Return reduced shears and magnification.
+        g1 = gamma1/(1. - kappa)    # real part of reduced shear
+        g2 = gamma2/(1. - kappa)    # imaginary part of reduced shear
+        mu = 1./((1. - kappa)**2 - (gamma1**2 + gamma2**2)) # magnification
+        return g1, g2, mu
+
+    def getDust(self, skycat_obj):
+        """
+        Return the extinction parameters for the skyCatalog object
+        corresponding to the specified index.
+
+        Parameters
+        ----------
+        skycat_obj : skycatalogs.BaseObject
+            The skycatalogs object, e.g., star or galaxy, which can
+            have several subcomponents, e.g., disk, bulge, knots.
+
+        Returns
+        -------
+        (internal_av, internal_rv, galactic_av, galactic_rv)
+        """
+        # For all objects, internal extinction is already part of the SED,
+        # so Milky Way dust is the only source of reddening.
+        internal_av = 0
+        internal_rv = 1.
+        MW_av_colname = f'MW_av_lsst_{self.band}'
+        galactic_av = skycat_obj.get_native_attribute(MW_av_colname)
+        galactic_rv = skycat_obj.get_native_attribute('MW_rv')
+        return internal_av, internal_rv, galactic_av, galactic_rv
+
+    def get_gsobject(self, skycat_obj, component, gsparams, rng, exp_time):
+        """
+        Return a galsim.GSObject for the specifed skyCatalogs object
+        and component.
+
+        Parameters
+        ----------
+        skycat_obj : skyCatalogs.BaseObject
+            A skyCatalogs object, e.g., a star or galaxy.
+        component : str
+            The name of the sub-component of the skyCatalogs object
+            to consider.
+        """
+        sed, magnorm = self.getSED_info(skycat_obj, component)
+        if sed is None or magnorm >= 50:
+            return None
```

Comment:
This is probably where we want to do the faint check.  We may not be able to use realized_flux anymore, since that's currently a non-trivial calculation using the (multiple) SEDs, so we may need to base it on magnorm I guess?  But with some definition of "faint" here, we can use the trivial sed for any faint component.